### PR TITLE
fix: BraceLabel not passing **kwargs to Brace

### PR DESCRIPTION
**Pull request by KingAndCross**
## Overview: What does this pull request change?

This PR fixes a bug in which the you can't pass arguments to the Brace constructor. My solution imitates the Axes approach, adding a parameter (brace_config) for the arguments to be passed to Brace.

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

File: manim/mobject/svg/brace.py [hunk 1]
```diff
@@ -156,6 +156,25 @@ def get_direction(self):
...
+        label text
+    brace_direction:vector default=DOWN
+    label_constructor default=MathTex,
+    font_size: float default=DEFAULT_FONT_SIZE,
+    buff: int, default=0.2
+        buff between label and obj
+    brace_config: dict, optional
+        Arguments to be passed to :class:`Brace`.
+    kwargs
+        Additional arguments to be passed to :class:`~.VMobject`.
```
Comment (behackl) [inline manim/mobject/svg/brace.py:175]: Some improvements to the documentation of some of the parameters.

```suggestion
    obj
        The mobject adjacent to which the brace is placed.
    text
        The label text.
    brace_direction
        The direction of the brace. By default ``DOWN``.
    label_constructor
        A class or function used to construct a mobject representing
        the label. By default :class:`~.MathTex`.
    font_size
        The font size of the label, passed to the ``label_constructor``.
    buff
        The buffer between the mobject and thre brace.
    brace_config
        Arguments to be passed to :class:`.Brace`.
    kwargs
        Additional arguments to be passed to :class:`~.VMobject`.
```

File: manim/mobject/svg/brace.py [hunk 2]
```diff
@@ -164,6 +183,7 @@ def __init__(
         label_constructor=MathTex,
         font_size=DEFAULT_FONT_SIZE,
         buff=0.2,
+        brace_config: dict | None = None,
```
Comment (behackl) [inline manim/mobject/svg/brace.py:186]: Could you add type hints to the other arguments above too? `brace_direction`  is probably best typed as `np.ndarray`, until we have #3086.

File: manim/mobject/svg/brace.py [hunk 3]
```diff
@@ -173,7 +193,11 @@ def __init__(
         self.buff = buff
         if isinstance(obj, list):
             obj = self.get_group_class()(*obj)
-        self.brace = Brace(obj, brace_direction, buff, **kwargs)
+        if brace_config is None:
+            self.brace_config = {}
+        else:
+            self.brace_config = brace_config
+        self.brace = Brace(obj, brace_direction, buff, **self.brace_config)
```
Comment (behackl) [inline manim/mobject/svg/brace.py:200]: I don't think it is necessary to store the `brace_config` as an attribute of `BraceLabel`. What do you think?

```suggestion
        if brace_config is None:
            brace_config = {}
        self.brace = Brace(obj, brace_direction, buff, **brace_config)
```

**Review by behackl — CHANGES_REQUESTED** [review]
Thanks for your contribution! The change itself looks good to me. I left a couple of comments suggestions, please take a look!

Please leave a comment or re-request a review to let me know when I should look at this again.

**Comment by KingAndCross** [comment]
I accepted the changes and added the type hinting. I'm not sure about the hintins for obj though.
because it seems like you can pass a list of mobjects too and it's converted to a VGroup.

`        if isinstance(obj, list):
            obj = self.get_group_class()(*obj)
`
so should the type hint be `obj: Mobject | list[Mobject]` ?

File: manim/mobject/svg/brace.py [hunk 4]
```diff
@@ -156,24 +156,46 @@ def get_direction(self):
...
+        Arguments to be passed to :class:`.Brace`.
+    kwargs
+        Additional arguments to be passed to :class:`~.VMobject`."""
+
     def __init__(
         self,
-        obj,
-        text,
-        brace_direction=DOWN,
+        obj: Mobject,
```
Comment (KingAndCross) [inline manim/mobject/svg/brace.py:183]: I'm not sure about this type hint for obj. It seems like you can pass a list of mobjects too and it's converted to a VGroup in lines 174-175. So should the type hint be `obj: Mobject | list[Mobject] `?

Comment (behackl) [reply manim/mobject/svg/brace.py:183]: I think this is fine as is for now; we'll probably go over it again as soon as we start tightening the type hints across the library.

**Review by behackl — APPROVED** [review]
Thank you for your contribution!

**Review by naveen521kk — APPROVED** [review]
LGTM

**Pull request merged**